p2p/host/autorelay/options.go

```diff
@@ -0,0 +1,135 @@
+package autorelay
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+type config struct {
+	peerChan     <-chan peer.AddrInfo
+	staticRelays []peer.AddrInfo
+	// see WithMinCandidates
+	minCandidates int
+	// see WithNumCandidates
+	maxCandidates int
+	// Delay until we obtain reservations with relays, if we have less than minCandidates candidates.
+	// See WithBootDelay.
+	bootDelay time.Duration
+	// backoff is the time we wait after failing to obtain a reservation with a candidate
+	backoff time.Duration
+	// If we fail to obtain a reservation more than maxAttempts, we stop trying.
+	maxAttempts int
+	// Number of relays we strive to obtain a reservation with.
+	desiredRelays int
+}
+
+var defaultConfig = config{
+	minCandidates: 4,
+	maxCandidates: 20,
+	bootDelay:     3 * time.Minute,
+	backoff:       time.Hour,
+	maxAttempts:   3,
+	desiredRelays: 2,
+}
+
+// DefaultRelays are the known PL-operated v1 relays; will be decommissioned in 2022.
+var DefaultRelays = []string{
+	"/ip4/147.75.80.110/tcp/4001/p2p/QmbFgm5zan8P6eWWmeyfncR5feYEMPbht5b1FW1C37aQ7y",
+	"/ip4/147.75.80.110/udp/4001/quic/p2p/QmbFgm5zan8P6eWWmeyfncR5feYEMPbht5b1FW1C37aQ7y",
+	"/ip4/147.75.195.153/tcp/4001/p2p/QmW9m57aiBDHAkKj9nmFSEn7ZqrcF1fZS4bipsTCHburei",
+	"/ip4/147.75.195.153/udp/4001/quic/p2p/QmW9m57aiBDHAkKj9nmFSEn7ZqrcF1fZS4bipsTCHburei",
+	"/ip4/147.75.70.221/tcp/4001/p2p/Qme8g49gm3q4Acp7xWBKg3nAa9fxZ1YmyDJdyGgoG6LsXh",
+	"/ip4/147.75.70.221/udp/4001/quic/p2p/Qme8g49gm3q4Acp7xWBKg3nAa9fxZ1YmyDJdyGgoG6LsXh",
+}
+
+var defaultStaticRelays []peer.AddrInfo
+
+func init() {
+	for _, s := range DefaultRelays {
+		pi, err := peer.AddrInfoFromString(s)
+		if err != nil {
+			panic(fmt.Sprintf("failed to initialize default static relays: %s", err))
+		}
+		defaultStaticRelays = append(defaultStaticRelays, *pi)
+	}
+}
+
+type Option func(*config) error
+
+func WithStaticRelays(static []peer.AddrInfo) Option {
+	return func(c *config) error {
+		if len(c.staticRelays) > 0 {
+			return errors.New("can't set static relays, static relays already configured")
+		}
+		c.staticRelays = static
+		return nil
+	}
+}
+
+func WithDefaultStaticRelays() Option {
+	return WithStaticRelays(defaultStaticRelays)
```

Comment:
default static relays is empty.

Reply:
It's filled in `init`. We need to parse the strings first.

Reply:
ah ok